src/modules/keycloak-administration/domain/keycloak-user.service.ts

```diff
@@ -265,4 +270,105 @@ export class KeycloakUserService {
+            return { ok: false, error: new KeycloakClientError('Could not create group') };
+        }
+    }
+
+    public async createRole(role: string): Promise<Result<string, DomainError>> {
+        const kcAdminClientResult: Result<KeycloakAdminClient, DomainError> =
+            await this.kcAdminService.getAuthedKcAdminClient();
+
+        if (!kcAdminClientResult.ok) {
+            return kcAdminClientResult;
+        }
+
+        try {
+            const roleName: string = role;
+            const existingRole: RoleRepresentation | undefined = await kcAdminClientResult.value.roles.findOneByName({
+                name: roleName,
+            });
+
+            if (existingRole) {
+                this.logger.info(`Role already exists: ${roleName}`);
+                return { ok: false, error: new KeycloakClientError('Role name already exists') };
+            }
+
+            const roleRepresentation: RoleRepresentation = {
+                name: roleName,
+            };
+
+            const response: { roleName: string } = await kcAdminClientResult.value.roles.create(roleRepresentation);
+            this.logger.info(`Keycloak role created: ${roleName}`);
+            return { ok: true, value: response.roleName };
+        } catch (err) {
+            this.logger.error(`Could not create role, message: ${JSON.stringify(err)} `);
+            return { ok: false, error: new KeycloakClientError('Could not create role') };
+        }
+    }
+
+    public async addRoleToGroup(groupId: string, createdRole: string): Promise<Result<boolean, DomainError>> {
+        const kcAdminClientResult: Result<KeycloakAdminClient, DomainError> =
+            await this.kcAdminService.getAuthedKcAdminClient();
+
+        if (!kcAdminClientResult.ok) {
+            return kcAdminClientResult;
+        }
+
+        try {
+            const roleName: string = createdRole;
+            const role: RoleRepresentation | undefined = await kcAdminClientResult.value.roles.findOneByName({
+                name: roleName,
+            });
+
+            if (!role || !role.id || !role.name) {
+                this.logger.error(`Role not found or id/name is undefined for: ${roleName}`);
+                return { ok: false, error: new KeycloakClientError('Role not found or id/name is undefined') };
+            }
+
+            await kcAdminClientResult.value.groups.addRealmRoleMappings({
+                id: groupId,
+                roles: [
+                    {
+                        id: role.id,
+                        name: role.name,
+                    },
+                ],
+            });
+
+            this.logger.info(`Role ${roleName} added to group with ID: ${groupId}`);
+            return { ok: true, value: true };
+        } catch (err) {
+            this.logger.error(`Could not add role to group, message: ${JSON.stringify(err)}`);
+            return { ok: false, error: new KeycloakClientError('Could not add role to group') };
+        }
+    }
```

Comment:
Maybe move these into their own service? (keycloak-role/group-service)

Not really a part of users.